src/Workspaces/Remote/Core/Diagnostics/DiagnosticComputer.cs

```diff
@@ -19,11 +20,15 @@ internal class DiagnosticComputer
     {
         private readonly Project _project;
         private readonly Dictionary<DiagnosticAnalyzer, HashSet<DiagnosticData>> _exceptions;
+        private readonly IPerformanceTrackerService _performanceTracker;
 
         public DiagnosticComputer(Project project)
         {
             _project = project;
             _exceptions = new Dictionary<DiagnosticAnalyzer, HashSet<DiagnosticData>>();
+
+            // we only track performance from primary branch. all forked branch we don't care such as preview.
+            _performanceTracker = project.IsFromPrimaryBranch() ? project.Solution.Workspace.Services.GetService<IPerformanceTrackerService>() : null;
```

Comment:
💭 I have no idea what "branch" means in this context, and it was never documented in the code (this is an existing issue in the code and not something caused by this pull request). 😦

Reply:
it is referring Workspace.PrimaryBranchId (http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/Workspace.cs,475af1d299809887,references)